File: kbe/res/client/sdk_templates/unity/EntityComponent.cs
namespace KBEngine
{
	using UnityEngine; 
	using System; 
	using System.Collections; 
	using System.Collections.Generic;
	using System.Threading;

	/*
		实体组件模块基础类
	*/
	public class EntityComponent
	{
		public UInt16 entityComponentPropertyID = 0;
		public UInt16 componentType = 0;
		public Int32 ownerID = 0;
		public Entity owner = null;

		public virtual void onRemoteMethodCall(UInt16 methodUtype, MemoryStream stream)
		{
			// 动态生成
		}

		public virtual void onUpdatePropertys(UInt16 propUtype, MemoryStream stream, int maxCount)
		{
			// 动态生成
		}

		public virtual void callPropertysSetMethods()
		{
			// 动态生成
		}

		public virtual void createFromStream(MemoryStream stream)
		{
			componentType = (UInt16)stream.readInt32();
			ownerID = stream.readInt32();

			//UInt16 ComponentDescrsType;
			stream.readUint16();

			UInt16 count = stream.readUint16();

			if(count > 0)
				onUpdatePropertys(0, stream, count);
		}
	}
}

File: kbe/res/sdk_templates/client/ue4/Source/KBEnginePlugins/Engine/EntityComponent.cpp

#include "EntityComponent.h"
#include "KBEngine.h"
#include "KBDebug.h"
#include "MemoryStream.h"

namespace KBEngine
{

EntityComponent::EntityComponent():
entityComponentPropertyID(0),
componentType(0),
ownerID(0),
pOwner(NULL)
{
}

EntityComponent::~EntityComponent()
{
}

void EntityComponent::createFromStream(MemoryStream& stream)
{
	componentType = (uint16)stream.readInt32();
	ownerID = stream.readInt32();

	//UInt16 ComponentDescrsType;
	stream.readUint16();

	uint16 count = stream.readUint16();

	if(count > 0)
		onUpdatePropertys(0, stream, count);
}

}
